src/main/java/org/apache/commons/csv/pretty/PrettyPrinter.java

```diff
@@ -0,0 +1,172 @@
+        this.records = parser.getRecords();
+        this.style = style;
+        this.out = System.out;
+        this.spaces = calculateSpaces(this.records);
+    }
+
+    public PrettyPrinter(final CSVParser parser, final TableStyle style, final PrintStream out) throws IOException {
+        this.records = parser.getRecords();
+        this.style = style;
+        this.out = out;
+        this.spaces = calculateSpaces(this.records);
+    }
+
+    private int[] calculateSpaces(final List<CSVRecord> records) {
+        final int length = records.isEmpty() ? 0 : records.iterator().next().size();
+        final int[] spaces = new int[length];
+        for (int i = 0; i < length; i++) {
+            int max = Integer.MIN_VALUE;
+            for (final CSVRecord record : records) {
+                max = Math.max(record.get(i).length(), max);
+            }
+            spaces[i] = 2 + max;
+        }
+        return spaces;
+    }
+
+    public List<CSVRecord> getRecords() {
+        return this.records;
+    }
+
+    public TableStyle getStyle() {
+        return this.style;
+    }
+
+    private void printUpperLine(final boolean thick, final char joint) {
+        if (this.spaces.length == 0) {
+            return;
+        }
+        out.print(thick ? this.style.THICK_UPPER_LEFT_CORNER_JOINT : this.style.THIN_UPPER_LEFT_CORNER_JOINT);
+        final char line = thick ? this.style.THICK_HORIZONTAL_LINE : this.style.THIN_HORIZONTAL_LINE;
+        final StringBuilder sb = new StringBuilder();
+        for (final int sp : this.spaces) {
+            sb.append(StringUtils.repeat(line, sp));
+            sb.append(joint);
+        }
+        out.print(sb.substring(0, sb.length() - 1));
+        out.print(thick ? this.style.THICK_UPPER_RIGHT_CORNER_JOINT : this.style.THIN_UPPER_RIGHT_CORNER_JOINT);
+    }
+
+    private void printMiddleLine(final boolean thickBorder, final boolean thickLine) {
+        if (this.spaces.length == 0) {
+            return;
+        }
+        out.print(thickBorder ? (thickLine ? this.style.THICK_THICK_LEFT_JOINT : this.style.THICK_LEFT_JOINT) : this.style.THIN_LEFT_JOINT);
+        final char line = thickLine ? this.style.THICK_HORIZONTAL_LINE : this.style.THIN_HORIZONTAL_LINE;
+        final char joint = thickLine ? this.style.THICK_MIDDLE_JOINT : this.style.THIN_MIDDLE_JOINT;
+        final StringBuilder sb = new StringBuilder();
+        for (final int sp : this.spaces) {
+            sb.append(StringUtils.repeat(line, sp));
+            sb.append(joint);
+        }
+        out.print(sb.substring(0, sb.length() - 1));
+        out.print(thickBorder ? (thickLine ? this.style.THICK_THICK_RIGHT_JOINT : this.style.THICK_RIGHT_JOINT) : this.style.THIN_RIGHT_JOINT);
+    }
+
+    private void printLowerLine(final boolean thick) {
+        if (this.spaces.length == 0) {
+            return;
+        }
+        out.print(thick ? this.style.THICK_LOWER_LEFT_CORNER_JOINT : this.style.THIN_LOWER_LEFT_CORNER_JOINT);
```

Comment:
I thought the point of the style class what to abstract out this constant testing of "thick". The constant testing of "thick" seems like a code smell and not very OO...